dashboard/app/views/layouts/_header.html.haml

```diff
@@ -164,6 +164,7 @@
 - elsif level
   :javascript
     dashboard.header.buildProjectInfoOnly()
+    dashboard.lab.init("#{@level.id}");
```

Comment:
Hmm, this does feel a little weird to me...want to make sure I understand correctly though - is this here because on level pages (/levels/<id>), the `currentLevelId` does not exist in progress redux? If that is the case, I think I'm what I don't feel great about is that we're using the presence of this field in labRedux as evidence that we're on a level page, i.e. baking in some assumptions about how this field was set. It seems like this also puts the burden of determining which situation we're in on the lab (MusicView), when this is really lab-agnostic (right?). Could we instead set this field on progress redux, along with some flag or information about whether we're on a script or level page?

Also, out of curiosity - do we get different level data back from the server if we hit the levels/<id>/level_data endpoint vs the /s/<script>/<lesson>/<levels>/<id>/level_data endpoint?

Reply:
1. After our discussion I've started using `currentLevelId` in `progressRedux` for level pages, and we'll see how that goes.
2. It's the same data so far, but I think worth having separate server APIs so that they can have different permission-checking, different caching, and potentially different results going forward.